modules/gcp/gke/main.tf

```diff
@@ -149,4 +149,20 @@ resource "google_project_iam_member" "service_account-roles" {
   project = var.gcp["project"]
   role    = each.value
   member  = "serviceAccount:${google_service_account.gke_service_account.email}"
-}
\ No newline at end of file
+}
+
+resource "google_compute_firewall" "cert-manager-webhook-firewall" {
+  name = "${var.region}-${var.environment}-cert-manager-webhook-firewall-rule"
+  network = var.network
```

Comment:
Is there some way to have this use the module self_link for network if var.network isn't defined?

Reply:
Is this to make the `gke` module independent of the `vpc` network output? When using the `vpc` module, we are already using the self_link for network:

https://github.com/cloudkite-io/terraform-modules/blob/master/modules/gcp/vpc/outputs.tf#L3

In the case of changing this `var.network` to use module self_link for network the only option is data source, although we will still need the network name.

Reply:
We don't need to use the data source if the name of the self_link is deterministic based on the name value input given to the network module. We generate the name for the network, so there's no reason we can't guess the name of the network if it's not specified.

Reply:
Changes already tested in dev and pushed.